gk/main.c

```diff
@@ -534,13 +609,10 @@ gk_proc(void *arg)
 				/* Drop non-IP packets. */
 				drop_packet(pkt);
 				continue;
-			} else if (pkt_is_nd(&packet, &gk_conf->net->front)) {
-				/*
-				 * TODO Use DPDK packet classification
-				 * and distribution here instead.
-				 */
-				if (submit_nd(pkt, &gk_conf->net->front) == -1)
-					drop_packet(pkt);
+			} else if (packet.flow.proto == ETHER_TYPE_IPv6 &&
+				     ipv6_if_configured(&gk_conf->net->front)) {
```

Comment:
I think we should drop the second test. The functional blocks helping the IPv6 filter to work should make the minimum assumption on what's going on in the filter.